lib/excon/middlewares/idempotent.rb

```diff
@@ -20,6 +20,9 @@ def error_call(datum)
 
         if datum[:idempotent] && [Excon::Errors::Timeout, Excon::Errors::SocketError,
             Excon::Errors::HTTPStatusError].any? {|ex| datum[:error].kind_of?(ex) } && datum[:retries_remaining] > 1
+
+          sleep(datum[:retry_interval]) if datum[:retry_interval] && !datum[:mock]
```

Comment:
Agreed that timing is tests are a pain. :) And mixing mock behaviour into real code is not a great trade off. Removing the mock and send a 0 internal seems reasonable to me.

Alternatively, what would you think about stubbing #sleep, and checking that it is execute the right number of times?
This would be more like RSpec so it might make sense to convert this test first.